hoomd/hpmc/pair/step.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2009-2024 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Step function pair potential.
+
+.. invisible-code-block: python
+
+    simulation = hoomd.util.make_example_simulation()
+    sphere = hoomd.hpmc.integrate.Sphere()
+    sphere.shape['A'] = dict(diameter=0.0)
+    simulation.operations.integrator = sphere
+"""
+
+import hoomd
+
+from .pair import Pair
+from hoomd.data.typeconverter import OnlyIf, to_type_converter
+
+
+@hoomd.logging.modify_namespace(('hpmc', 'pair', 'Step'))
+class Step(Pair):
```

Comment:
It may be helpful to add a note in the docs saying the r values need to be sorted by distance or else this will not work right.

Reply:
We also need to validate this condition to prevent errors.